### ISSUE TITLE
Question: Discriminator.py versus Classifier.py Deepfake Detection Accuracy Upper Bound?

### ISSUE THREAD

**Issue by MartialTerran** [NVlabs/stylegan3#660]
**Title:** `[Question/Discussion] Deepfake Detection Upper Bound: Why do our CNN-based detectors plateau at ~95% Accuracy on StyleGAN3 outputs?`

Hello StyleGAN3 Team,

First, thank you for the incredible work on this architecture. The alias-free, equivariant design is a masterclass in generative modeling.

NVIDIA has invited researches to use "fake" images generated by its StyleGAN3 to improve Real/Fake Detectors (Classifiers). 
https://blogs.nvidia.com/blog/how-researchers-use-nvidia-ai-to-help-mitigate-misinformation/

This [building Real/Fake Detectors (Classifiers)] is one aspect of what we are doing now. 

I am reaching out with a research question regarding the detectability of StyleGAN3 "fake" outputs. We are currently using a Real/Fake dataset published on Kaggle [ /kaggle/input/datasets/chuneeb/deepfake-detection-dataset-2026/FINAL_DATASET.csv ] and training heavily optimized Vision Transformers and CNN-hybrid models on the dataset consisting of about 6k of Real vs. StyleGAN3-generated Fake images. The dataset labels indicate that the origin of all the "fake" images is "StyleGAN3" but the images are sized at "224x224" (which is not a native output resolution of the StyleGAN models).  

We have built and trained several model architectures that train to a Train Accuracy of about 95% and a maximum Test Accuracy of 95.1%
For example:
  Ep 1/12 | Train Loss: 0.189 | Test Acc: 89.0%
  Ep 2/12 | Train Loss: 0.178 | Test Acc: 94.9%
  Ep 3/12 | Train Loss: 0.170 | Test Acc: 94.9%
  Ep 4/12 | Train Loss: 0.159 | Test Acc: 94.9%
  Ep 5/12 | Train Loss: 0.157 | Test Acc: 94.5%
  Ep 6/12 | Train Loss: 0.159 | Test Acc: 94.9%
  Ep 7/12 | Train Loss: 0.162 | Test Acc: 94.5%
  Ep 8/12 | Train Loss: 0.152 | Test Acc: 94.7%
  Ep 9/12 | Train Loss: 0.164 | Test Acc: 95.1%
  Ep 10/12 | Train Loss: 0.153 | Test Acc: 93.2%
  Ep 11/12 | Train Loss: 0.157 | Test Acc: 94.7%
  Ep 12/12 | Train Loss: 0.165 | Test Acc: 94.9%

The likely culprit (depriving us of 99% Accuracy) is **StyleGAN3's alias-free architecture**? Older GANs (like StyleGAN1 and 2) left behind high-frequency "checkerboard" artifacts and structural noise due to how convolutions and upsampling were matheatically applied? Deepfake detectors could easily hit 99.9% accuracy by just looking at the frequency domain? **StyleGAN3 explicitly removed these artifacts.** By enforcing strict translation and rotation equivariance, StyleGAN3 fakes are notoriously missing the typical "CNN fingerprints" that standard deepfake detectors rely on?

**Our Observation:**
No matter how we scale the architecture or optimize the hyperparameters, our Train and Test Accuracies firmly plateau at around **95.1%**.  We suppose that the dataset contains images originally generated at 1024x1024 by StyleGAN3 but then somehow scaled down to 224x224.  Should we use a different real/fake dataset and re-train and expect different (Better/Worse Accuracy) results? Real/Fake Dataset recommendations?]

**Our Questions:**
1. **The Detection Ceiling:** In older architectures (like StyleGAN2), simple CNNs could easily hit 99%+ accuracy by latching onto high-frequency upsampling artifacts or phase/frequency discrepancies? Given that StyleGAN3 was specifically engineered to be alias-free and suppress these exact structural artifacts, is a ~95% detection accuracy upper-bound consistent with what your team (or the broader forensics community) has observed when training binary classifiers strictly on StyleGAN3 outputs and same number of "Real" images?  [Are we doing better or worse than others?]
2. **Discriminator Insights:** While we understand that at GAN equilibrium the internal Discriminator operates near 50% accuracy, have you experimented with taking a fully-trained StyleGAN3 Discriminator, freezing the Generator, and fine-tuning the Discriminator purely as a Real/Fake Classifier? If so, does a Classifier also hit a similar hard ceiling around 95% or finish at 50%?   Any insights into the statistical fingerprints (or lack thereof) of StyleGAN3 that cause our state-of-the-art classifiers to hit this specific ceiling would be immensely appreciated. 
3. **Downscaling Effects:** Do you believe scaling the 1024x1024 images down to 224x224 is destroying the remaining high-frequency phase discrepancies that a detector might use to close that final 5% gap?

Extended comments:
The standard StyleGAN3 Discriminator operates at 50%, but I am asking what happens if the Discrimimator is replaced with one of our 95% Classifiers, and the whole model is then fine-tuned with that classifier?  Will this tend to alter the StyleGAN model output so as make our Classifier's Accuracy fall to 50% and thus its utility as a real/fake Detector will be nullified?  Or....?   In a perfectly trained Generative Adversarial Network (GAN), the Discriminator and Generator reach a *Nash Equilibrium*. This means the Generator becomes so good at faking images that the Discriminator is forced into completely random guessing (50% accuracy)? Therefore, the pre-trained Discriminator inside the released StyleGAN3 model does *not* have 95%+ accuracy; its accuracy on the final generated images is actually hovering around 50% by mathematical design?  But then why is our custom-trained classifier plateauing at 95.1% (not 50%) on "StyleGAN3" images?**

Downscaling an image (especially with standard bilinear/bicubic interpolation) acts as a low-pass filter? This might actually be destroying much of the evidence our models need to get beyond 95% (e.g., to 99%)?  Does resizing 1kx1k images to 224x224 make 5% of fakes undetectable in the broader field of Fake-Detection known to NVIDIA and the researchers using StyleGAN3 images datasets?  Why 5%?  Or, why "only %5" ?

Thank you for your time and for advancing the field!

Martial Terran 
https://huggingface.co/MartialTerran

P.S.  We have run the pretrained StyleGAN3 model 
!git clone https://github.com/NVlabs/stylegan3.git
url = "https://api.ngc.nvidia.com/v2/models/nvidia/research/stylegan3/versions/1/files/stylegan3-r-ffhqu-1024x1024.pkl"
(on a Colab T4 GPU) to generate images at 1024x1024 resolution.  Thus, we can generate "fake" Images from StyleGAN3-R etc, but this does not provide me with an appropriate set of similar "real images" for purposes of training our Real/Fake Classifier models. 

Notes:
To:
[PDillis](https://github.com/PDillis)
[Diego Porres](https://github.com/PDillis)

Our Real/Fake Classifier project focuses on detecting AI-generated human faces using modern computer vision techniques.

🔥 Project Goals Detect REAL vs FAKE faces 
Build a robust deepfake classifier to Analyze and Classify images. 
Visualize model predictions using Explainable AI 
Deploy a real-world inference pipeline

🚀 Applications Social media verification AI-generated media detection Cybersecurity Digital identity protection

The dataset on Kaggle supposedly contains both authentic and synthetic face images collected from real-world and GAN-generated sources.  Along with image data, the dataset includes rich metadata such as age group, gender, confidence score, image quality, and detection difficulty.

📊 Dataset Features 6,557 face images REAL and FAKE labels Difficulty ratings Confidence scores Gender and age metadata Predefined train/test/validation split: [ /kaggle/input/datasets/chuneeb/deepfake-detection-dataset-2026/FINAL_DATASET.csv ]
[supposedly obtained from StyleGAN3 https://nvlabs.github.io/stylegan3/ ] [ https://www.youtube.com/watch?v=Vuz-cRa1Ke0&t=4s ] [ https://github.com/NVlabs/stylegan3 ] 1024x1024   
[We are only using the Real/Fake lable for training (classification) purposes in our models]    

StyleGAN3 - Rotation and Translation! [ https://www.youtube.com/watch?v=Vuz-cRa1Ke0&t=4s ] == Covered in this video == What's new in StyleGAN3? Requirements How to download StyleGAN3 Setting up your Python virtual environment Dataset creation Getting help with commands Network training Generating images with StyleGAN3 Generating interpolated videos with StyleGAN3 using seeds Exploring StyleGAN3 networks using the visualiser CLIP guided StyleGAN3 Audio-reactive StyleGAN3

GitHub: https://github.com/nerdyrodent/stylegan3

Buy art - support a nerd :) https://www.hicetnunc.xyz/NerdyRodent
